outsource/config.py

```diff
@@ -301,15 +123,37 @@ def get_requirements(self, rses):
         return requirements, requirements_us
 
     def depends_on(self, data_type):
-        return DEPENDS_ON[data_type]
+        """Get the data_type this one depends on.
+
+        The result will be either data_type in same level of PER_CHUNK_DATA_TYPES or the
+        root_data_types.
+
+        """
+        possible_dependencies = get_possible_dependencies(self.context)
+        # Get the highest data_type in the possible dependencies
+        data_types = sorted(
+            self.context.get_dependencies(data_type) & (possible_dependencies),
```

Comment:
Why is self.context.get_dependencies(data_type) not sufficient? Why do we need & operation?

Reply:
Here the `RunConfig.depends_on` returns data_type in `PER_CHUNK_DATA_TYPES` or `raw_records*`. So the `&` here limits the range of searching. The `RunConfig.depends_on` is not the same as `depends_on` in strax, but is a guide on which important data_type we should look at when checking the available RSE. You can search for the usage of `RunConfig.depends_on` to look at the details.